llvm/lib/Transforms/Vectorize/LoopIdiomVectorize.cpp

```diff
@@ -717,12 +703,8 @@ void LoopIdiomVectorize::transformByteCompare(GetElementPtrInst *GEPA,
   if (FoundBB != EndBB) {
     Value *FoundCmp = Builder.CreateICmpEQ(ByteCmpRes, MaxLen);
     Builder.CreateCondBr(FoundCmp, EndBB, FoundBB);
-    DTU.applyUpdates({{DominatorTree::Insert, CmpBB, FoundBB},
```

Comment:
Given that FoundBB and EndBB appear after CmpBB and we're only calling `DTU.applyUpdates({{DominatorTree::Insert, MismatchEnd, CmpBB}});` above I'm not sure how the dominator tree gets updated about the new paths CmpBB->FoundBB and CmpBB->EndBB? I admit my knowledge of the DTU isn't that great, so I'm probably missing something!